var/spack/repos/builtin/packages/py-nilearn/package.py

```diff
@@ -53,29 +56,10 @@ class PyNilearn(PythonPackage):
     depends_on("py-matplotlib@1.1.1:3.3", when="@:0.6 +plotting", type=("build", "run"))
 
     @property
-    def import_modules(self):
-        modules = [
-            "nilearn",
-            "nilearn.connectome",
-            "nilearn.datasets",
-            "nilearn.decoding",
-            "nilearn.decomposition",
-            "nilearn.image",
-            "nilearn.input_data",
-            "nilearn.masking",
-            "nilearn.mass_univariate",
-            "nilearn.regions",
-            "nilearn.signal",
-            "nilearn.surface",
-        ]
+    def skip_modules(self):
+        modules = []
 
-        if self.spec.satisfies("@0.6:"):
-            modules.append("nilearn.externals")
-
-        if self.spec.satisfies("@0.7:"):
-            modules.append("nilearn.glm")
-
-        if "+plotting" in self.spec:
+        if self.spec.satisfies("~plotting"):
             modules.append("nilearn.plotting")
             if self.spec.satisfies("@0.7:"):
                 modules.append("nilearn.reporting")
```

Comment:
Is `nilearn.reporting` installed with nilearn 0.6? If not, you can remove this, as `import_modules` won't auto-detect something that doesn't exist, so there's nothing to skip.

If yes, does `nilearn.reporting` require `+plotting`? If not, this shouldn't be indented under the same if-statement.

Reply:
Yes, `nilearn.reporting` is installed with `nilearn@0.6` but does not require a `matplotlib` dependency. Starting from `nilearn@0.7`, `nilearn.reporting` requires `matplotlib` to work.